DataStructures/Trees/BinarySearchTree.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Text;
+
+
+namespace DataStructures.Trees
+{
+    public class BinarySearchTree
```

Comment:
You should technically have a separate class called "BinaryTree" where you implement your PreOrder, InOrder, and PostOrder, as well as an Add() that simply just adds a node to the left or the right if either is null (as opposed to how your add is setup below which is sorting as a SearchTree should).  It will simply say while(current != null) { if (current.Left == null) { current.Left = new value; return; } else if (current.Right == null { current.Right = value; return; } else { traverse until you find a null value to stub in your new value } } or something along those lines.

Reply:
And then, your BinarySearchTree will inherit from your BinaryTree and have access to those three traversals as well as it's own implementation of Add() for the sake of sorting accordingly.